src/resource.ts

```diff
@@ -16,8 +17,10 @@ const { MONGOOSE_CAST_ERROR, MONGOOSE_DUPLICATE_ERROR_CODE, MONGOOSE_VALIDATION_
  * Adapter for mongoose resource
  * @private
  */
-class Resource extends BaseResource {
-  private readonly dbType: string = 'mongodb'
+export abstract class Resource extends BaseResource {
+  public abstract options: IAdapterOptions;
+
+  public readonly dbType: string = 'mongodb'
```

Comment:
Note had to change this to public as otherwise would get the TS error:
```
error TS4094: Property 'connection' of exported class expression may not be private or protected.
```

Reply:
I managed to make it protected